packages/signature-controller/src/types.ts

```diff
@@ -71,15 +77,41 @@ export type MessageParamsTyped = MessageParams & {
         primaryType: string;
         message: Json;
       };
-
   /** Version of the signTypedData request. */
   version?: string;
 };
 
+/** Information about various state changes returned by decoding api. */
+type DecodingDataStateChanges = {
+  assetType: string;
+  changeType: string;
+  address: string;
+  amount: string;
+  contractAddress: string;
+}[];
+
+/** Error details for unfulfilled the decoding request. */
+type DecodingDataError = {
+  message: string;
```

Comment:
One for the API potentially, but is it more flexible to return error codes to let the UI generate suitable messages and translate them?